### PR TITLE
tracking: Support create parameters in python bindings

### PR DESCRIPTION
**Pull request by ledmonster**
### This pullrequest changes

This PR add support tracking parameters to python bindings.

I'm not familiar with OpenCV's python binding, and I wrote this PR based on following document and SimpleBlobDetector's implementation.

- https://docs.opencv.org/3.4/da/d49/tutorial_py_bindings_basics.html

Any feedback is welcome. Thank you.

### CODE REVIEW

File: modules/tracking/include/opencv2/tracking/tracker.hpp [hunk 1]
```diff
@@ -900,15 +900,15 @@ class CV_EXPORTS TrackerSamplerPF : public TrackerSamplerAlgorithm
...
-    double alpha; //!< with each new round we exponentially decrease the amount of "perturbing" we allow (like in simulated annealing)
-                  //!< and this very alpha controls how fast annealing happens, ie. how fast perturbing decreases
-    Mat_<double> std; //!< initial values for perturbing (1-by-4 array, as each rectangle is given by 4 values -- coordinates of opposite vertices,
-                      //!< hence we have 4 values to perturb)
+    CV_WRAP Params();
+    CV_PROP_RW int iterationNum; //!< number of selection rounds
+    CV_PROP_RW int particlesNum; //!< number of "perturbed" boxes on each round
+    CV_PROP_RW double alpha;     //!< with each new round we exponentially decrease the amount of "perturbing" we allow (like in simulated annealing)
+                                 //!< and this very alpha controls how fast annealing happens, ie. how fast perturbing decreases
+    Mat_<double> std;            //!< initial values for perturbing (1-by-4 array, as each rectangle is given by 4 values -- coordinates of opposite vertices,
```
Comment (ledmonster) [inline modules/tracking/include/opencv2/tracking/tracker.hpp]: I'm not sure how to support `Mat_` in python bindings.

**Comment by ledmonster** [comment]
Java bindings build failed. Umm...

File: modules/tracking/include/opencv2/tracking/tracker.hpp [hunk 2]
```diff
@@ -780,15 +780,15 @@ class CV_EXPORTS TrackerSamplerCSC : public TrackerSamplerAlgorithm
...
+  struct CV_EXPORTS_W_SIMPLE Params
   {
-    Params();
-    float initInRad;        //!< radius for gathering positive instances during init
-    float trackInPosRad;    //!< radius for gathering positive instances during tracking
-    float searchWinSize;  //!< size of search window
-    int initMaxNegNum;      //!< # negative samples to use during init
-    int trackMaxPosNum;     //!< # positive samples to use during training
-    int trackMaxNegNum;     //!< # negative samples to use during training
+    CV_WRAP Params();
```
Comment (alalek) [inline modules/tracking/include/opencv2/tracking/tracker.hpp]: > CV_WRAP

Remove from this line.

---

Similar is below.

Comment (ledmonster) [reply modules/tracking/include/opencv2/tracking/tracker.hpp]: @alalek thank you. It works. We can still use default constructor in Python.

File: modules/tracking/include/opencv2/tracking/tracker.hpp [hunk 3]
```diff
@@ -1496,44 +1484,42 @@ class CV_EXPORTS_W TrackerCSRT : public Tracker
...
-
-    float psr_threshold; //!< we lost the target, if the psr is lower than this.
+    CV_PROP_RW bool use_hog;
+    CV_PROP_RW bool use_color_names;
+    CV_PROP_RW bool use_gray;
+    CV_PROP_RW bool use_rgb;
+    CV_PROP_RW bool use_channel_weights;
+    CV_PROP_RW bool use_segmentation;
+
+    CV_PROP_RW std::string window_function; //!<  Window function: "hann", "cheb", "kaiser"
```
Comment (ledmonster) [inline modules/tracking/include/opencv2/tracking/tracker.hpp]: I added converters in `pyopencv_tracking.hpp` for this line.

**Comment by ledmonster** [comment]
Sorry, I didn't realize that till now, but tracking module already supported parameter via FileStorage API. So this PR should be revoked.

```
import cv2

# write
tracker = cv2.TrackerMedianFlow_create()
tracker.save('params.json')

# write (another way)
fs = cv2.FileStorage("params.json", cv2.FileStorage_WRITE)
tracker.write(fs)
fs.release()

# read
tracker2 = cv2.TrackerMedianFlow_create()
fs = cv2.FileStorage("params.json", cv2.FileStorage_READ)
tracker2.read(fs.getNode('my_object'))
```

**Pull request closed without merging**